Automattic/pocket-casts-ios
# Episodes Autoplay: persist from where the episode started playing

Adds logic so the app can know from where an episode started playing and persist it.

## To test

This PR doesn't have any user visible changes, so in order to test add this line to `AutoplayHelper.swift:33`:

```swift
print("🎧 played from: \(playlist)")
```

And this line to `AppDelegate.swift:86`:

```swift
print("🎧 latest played from: \(AutoplayHelper.shared.lastPlaylist)")
```

Then:

1. Run the app
2. Go to Podcasts tab, select any podcast and play an episode by tapping the play button at the right of the episode name
3. ✅ You should see `🎧 played from: podcast(uuid: UUID)` in the console (`UUID` being the podcast UUID from the episode you are playing)
3. Tap any other episode so the episode detail opens and tap play
3. ✅ You should see `🎧 played from: podcast(uuid: UUID)` in the console (`UUID` being the podcast UUID from the episode you are playing)
4. Stop the app and run it again
5. ✅ `🎧 latest played from: Optional(podcasts.EpisodesDataManager.Playlist.podcast(uuid: UUID))` in the console (`UUID` being the podcast UUID from the episode you are playing)
6. Repeat the steps above for: filters (filter tab), Downloads, Files, and Starred

You can also try Siri shortcuts,  Now Playing widget and Listening History: those should not be tracked.

## Checklist
